### PR TITLE
Feature: impl rename_database with kv-txn

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Feature: impl rename_database with kv-txn

## Changelog

- New Feature

## Related Issues

Fixes [Issue #5192 ](https://github.com/datafuselabs/databend/issues/5192)

### CODE REVIEW

File: metasrv/src/executor/meta_handlers.rs [hunk 1]
```diff
@@ -143,6 +146,35 @@ impl RequestHandler<DropDatabaseReq> for ActionHandler {
     }
 }
 
+#[async_trait::async_trait]
+impl RequestHandler<RenameDatabaseReq> for ActionHandler {
```
Comment (BohuTANG) [inline metasrv/src/executor/meta_handlers.rs:150]: This seems not a KV API. Is it the old-way meta API?

File: common/meta/grpc/src/meta_client_on_kv_impl.rs [hunk 2]
```diff
@@ -194,6 +196,64 @@ impl MetaApi for MetaClientOnKV {
...
+
+            {
+                let new_tenant_dbname = DatabaseNameIdent {
+                    tenant: tenant_dbname.tenant.clone(),
+                    db_name: req.new_db_name.clone(),
+                };
+                let txn_req = TxnRequest {
+                    condition: vec![
+                        self.txn_cond_seq(tenant_dbname, Eq, db_id_seq)?,
+                        self.txn_cond_seq(&db_id_key, Eq, db_meta_seq)?,
```
Comment (drmingdrmer) [inline common/meta/grpc/src/meta_client_on_kv_impl.rs:232]: Maybe a `rename` does not need to be blocked by `db_meta` changes.

**Review by drmingdrmer — COMMENTED** [review]
When renaming a db, it has to assert the **dest** db-name does not exist.

BTW, this is not a emergency task for now, it is in the **postponed task** list. Meanwhile the first to do is to replace the old MetaApi impls with the kv-txn based impl. After that, this PR will be simpler to impl too

File: metasrv/src/executor/meta_handlers.rs (hunk 1 above)
Comment (lichuang) [reply metasrv/src/executor/meta_handlers.rs:150]: > This seems not a KV API. Is it the old-way meta API?

After all meta API has impl by KV api, all the meta handlers in old way will be removed. But now, in order to pass compile, it has to impl with an old way API.

**Comment by lichuang** [comment]
> this is not a emergency task for now, it is in the postponed task list. 

Ok.

**Pull request closed without merging**